controllers/cronset_controller_test.go

```diff
@@ -83,98 +84,78 @@ var expectedOwnerRefs = []metav1.OwnerReference{{
 	Controller: &trueVal, BlockOwnerDeletion: &trueVal,
 }}
 
-func TestCronSetEvent_CreateCronJob(t *testing.T) {
-	ctx := context.Background()
-	scheme, err := batchv1alpha1.SchemeBuilder.Build()
-	require.NoError(t, err)
-	require.NoError(t, corev1.SchemeBuilder.AddToScheme(scheme))
-	require.NoError(t, batchv1.SchemeBuilder.AddToScheme(scheme))
-
-	t.Run("When creating a CronSet object", func(t *testing.T) {
-		fakeClient := fake.NewClientBuilder().WithScheme(scheme).WithObjects(node).WithObjects(cronSet).Build()
-		reconciler := CronSetReconciler{
-			fakeClient,
-			ctrl.Log.WithName("controllers").WithName("CronSet"),
-			scheme,
-		}
-
-		_, err = reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: cronSetKey})
-		assert.NoError(t, err)
-
-		t.Run("Should create a CronJob object into the relevant nodes", func(t *testing.T) {
-			createdCronJob := &batchv1.CronJob{}
-			err = fakeClient.Get(ctx, nodeCronJobKey, createdCronJob)
-			assert.NoError(t, err)
-			assert.NotEmpty(t, createdCronJob)
-			assert.Equal(t, expectedOwnerRefs, createdCronJob.OwnerReferences)
-		})
-	})
-
+type CronSetSuite struct {
+	suite.Suite
+	reconciler CronSetReconciler
+	fakeClient client.Client
 }
 
-func TestCronSetEvent_UpdateCronJob(t *testing.T) {
-	ctx := context.Background()
+func (s *CronSetSuite) SetupTest() {
+	ctx = context.Background()
 	scheme, err := batchv1alpha1.SchemeBuilder.Build()
-	require.NoError(t, err)
-	require.NoError(t, corev1.SchemeBuilder.AddToScheme(scheme))
-	require.NoError(t, batchv1.SchemeBuilder.AddToScheme(scheme))
+	require.NoError(s.T(), err)
+	require.NoError(s.T(), corev1.SchemeBuilder.AddToScheme(scheme))
+	require.NoError(s.T(), batchv1.SchemeBuilder.AddToScheme(scheme))
+
+	s.fakeClient = fake.NewClientBuilder().WithScheme(scheme).WithObjects(node).WithObjects(cronSet).Build()
```

Comment:
Each time the TC runs, the `SetupTest()` method is called, initializing the `fakeclient`. Because of this, the CronSet created by the previous TC is not in `fakeClient.tracker.objects`, which is causing the `fakeClient.Get(...)` call to fail.

Reply:
I added fakeClient.Create( cronset.. ) in a TC. but when I call Get func, it was not founded. please test in ur local env or I will show you tomorrow. :(

Reply:
ah... cronSet object passed as an argument to `fakeClient.Create(...)` is being passed to the address of a reference, causing fields like `resourceVersion` to be updated when it is created.
This seems to be the reason why the creation operation cannot be performed anymore.

I think We can use the values copied via `cronSet.DeepCopy()`.